transaction/LockManager.cpp

```diff
@@ -0,0 +1,235 @@
+        }
+      }
+    }
+    for (std::uint64_t tries = 0; tries < max_try_inner; ++tries) {
+      if (!inner_pending_requests_.empty()) {
+        LockRequest request = inner_pending_requests_.front();
+
+        bool result = acquireLock(request.getTransactionId(),
+                                  request.getResourceId(),
+                                  request.getAccessMode());
+        if (result) {
+          inner_pending_requests_.pop();
+          permitted_requests_.push(request);
+        }
+      }
+    }
+
+    killVictims();
+  }
+}
+
+bool LockManager::acquireLock(const transaction_id tid,
+                              const ResourceId &rid,
+                              const AccessMode access_mode) {
+  std::stack<std::pair<ResourceId, AccessMode>> stack;
+  ResourceId current_rid = rid;
+  AccessMode current_access_mode = access_mode;
+  stack.push(std::make_pair(current_rid, current_access_mode));
+
+  while (current_rid.hasParent()) {
+    current_rid = current_rid.getParentResourceId();
+    current_access_mode =
+      (current_access_mode.isShareLock() ||
+       current_access_mode.isIntentionShareLock())
+      ? AccessMode(AccessModeType::kIsLock)
+      : AccessMode(AccessModeType::kIxLock);
+
+    stack.push(std::make_pair(current_rid, current_access_mode));
+  }
+
+  lock_table_->latchExclusive();
+
+  while (!stack.empty()) {
+    std::pair<ResourceId, AccessMode> pair_to_pick = stack.top();
+    ResourceId rid_to_pick = pair_to_pick.first;
+    AccessMode access_mode_to_pick = pair_to_pick.second;
+
+    bool result = acquireLockInternal(tid,
+                                      rid_to_pick,
+                                      access_mode_to_pick);
+    if (!result) {
+      lock_table_->unlatchExclusive();
+      return false;
+    }
+    stack.pop();
+  }
+  lock_table_->unlatchExclusive();
+  return true;
+}
+
+bool LockManager::releaseAllLocks(const transaction_id tid) {
+  std::vector<ResourceId> related_rids
+      = transaction_table_->getResourceIdList(tid);
+  TransactionTableResult transaction_deleted
+      = transaction_table_->deleteTransaction(tid);
+  if (transaction_deleted
+          == TransactionTableResult::kTransactionDeleteError) {
+    FATAL_ERROR("In LockManager.releaseAllLocks"
+                " transaction could not be deleted!");
+  }
+
+  lock_table_->latchExclusive();
+
+  for (std::vector<ResourceId>::const_iterator it = related_rids.begin();
+       it != related_rids.end();
+       ++it) {
+    LockTableResult lock_deleted = lock_table_->deleteLock(tid, *it);
+
+    std::cout << "Transaction " + std::to_string(tid)
+      + " released lock:" + it->toString() + "\n";
+
+    if (lock_deleted == LockTableResult::kDelError) {
+      FATAL_ERROR("In LockManager.releaseAllLock "
+                  "lock could not be deleted from LockTable");
+    }
+  }
+
+  lock_table_->unlatchExclusive();
+  return true;
+}
+
+// If not blocked return true
+// if block return false
```

Comment:
Remove these comments. They should appear in the method declaration in the header file.